crawl-ref/source/spl-util.h

```diff
@@ -10,28 +10,30 @@
 #include "enum.h"
 #include "mon-info.h"
 
-enum spschool_flag_type
+enum class spschool
 {
-  SPTYP_NONE           = 0,
-  SPTYP_CONJURATION    = 1<<0,
-  SPTYP_HEXES          = 1<<1,
-  SPTYP_CHARMS         = 1<<2,
-  SPTYP_FIRE           = 1<<3,
-  SPTYP_ICE            = 1<<4,
-  SPTYP_TRANSMUTATION  = 1<<5,
-  SPTYP_NECROMANCY     = 1<<6,
-  SPTYP_SUMMONING      = 1<<7,
-  SPTYP_TRANSLOCATION  = 1<<8,
-  SPTYP_POISON         = 1<<9,
-  SPTYP_EARTH          = 1<<10,
-  SPTYP_AIR            = 1<<11,
-  SPTYP_LAST_SCHOOL    = SPTYP_AIR,
-  SPTYP_RANDOM         = SPTYP_LAST_SCHOOL << 1,
+  none           = 0,
+  conjuration    = 1<<0,
+  hexes          = 1<<1,
+  charms         = 1<<2,
+  fire           = 1<<3,
+  ice            = 1<<4,
+  transmutation  = 1<<5,
+  necromancy     = 1<<6,
+  summoning      = 1<<7,
+  translocation  = 1<<8,
+  poison         = 1<<9,
+  earth          = 1<<10,
+  air            = 1<<11,
+  LAST_SCHOOL    = spschool::air,
+  RANDOM         = spschool::LAST_SCHOOL << 1,
 };
-DEF_BITFIELD(spschools_type, spschool_flag_type, 11);
-const int SPTYP_LAST_EXPONENT = spschools_type::last_exponent;
-COMPILE_CHECK(spschools_type::exponent(SPTYP_LAST_EXPONENT)
-              == SPTYP_LAST_SCHOOL);
+DEF_BITFIELD(spschools_type, spschool, 11);
+const int SPSCHOOL_LAST_EXPONENT = spschools_type::last_exponent;
+COMPILE_CHECK(spschools_type::exponent(SPSCHOOL_LAST_EXPONENT)
+              == spschool::LAST_SCHOOL);
+// Moved from mapdef.cc:5318, needed to ensure randbook spschools are short ints
+COMPILE_CHECK(static_cast<int>(spschool::LAST_SCHOOL) < SHRT_MAX);
 
```

Comment:
I moved this `COMPILE_CHECK` over from an arbitrary place in `mapdef.cc` because I thought that having it in `mapdef.cc` and not right next to the enum class definition meant it could easily be missed when expanding this enum class.

Would you prefer it be here in `spl-util.h` (where it can easily be seen by anyone expanding the enum class) or in `mapdef.cc` (where its purpose is more obvious)?

(I have also marked the old position of this `COMPILE_CHECK` in this review.)

Reply:
I would prefer it here, so that it's not so easily missed. Might just be the GitHub interface but watch the wrap on the comment.

Reply:
The comment is exactly 80 characters long.

Reply:
That's what I get for eyeballing.